### PR TITLE
Fix #154 -- better log messages for do_patch()

### LINKED ISSUE #154: Misleading log message : 'Patched <package-name>'
When installing a package that can be conditionally patched the log message in `spack/package.py`:
```
tty.msg("Patched %s" % self.name)
```
is always printed, regardless of whether a patch has been applied or not. This may be misleading for someone that reads only the log messages and:
- doesn't expect the package to be patched
- doesn't know that the additional message:
```
tty.msg('Applying patch %s' % patch.path_or_url)
```
is logged if a patch is applied.

### PR DESCRIPTION
**Pull request by tgamblin**
@alalazo: can you take a look at this fix?  I haven't tested it rigorously but I believe it provides better patch semantics.  If you approve I'll merge.

This fixes #154

### CODE REVIEW

File: lib/spack/spack/package.py [hunk 1]
```diff
@@ -697,29 +698,46 @@ def do_patch(self):
         if os.path.isfile(good_file):
             tty.msg("Already patched %s" % self.name)
             return
+        elif os.path.isfile(no_patches_file):
```
Comment (alalazo) [inline lib/spack/spack/package.py]: I think it should read `no_patch_file`

File: lib/spack/spack/package.py [hunk 2]
```diff
@@ -697,29 +698,46 @@ def do_patch(self):
...
 
-        if has_patch_fun:
-            self.patch()
-
-        tty.msg("Patched %s" % self.name)
+        # touch good or no patches file so that we skip next time.
+        if patched:
+            touch(good_file)
+        else:
+            touch(no_patches_file)
```
Comment (alalazo) [inline lib/spack/spack/package.py]: Same typo here.

**Comment by tgamblin** [comment]
Will merge once I fix the bugs above and test it more thoroughly.

**Comment by tgamblin** [comment]
fixed, tested, merging.

**Pull request merged**